### PR TITLE
Add /item/import and deposit switch routes

### PR DESCRIPTION
**Pull request by jeffzwang**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -744,179 +801,290 @@ declare module 'plaid' {
...
+    ): void;
+    createDepositSwitchToken(
+      depositSwitchId: string,
+      cb: Callback<DepositSwitchTokenCreateResponse>,
+    ): void;
+    createDepositSwitchToken(
+      depositSwitchId: string,
+      options: Object,
+      cb: Callback<DepositSwitchTokenCreateResponse>,
     ): void;
```
Comment (jeffzwang) [inline index.d.ts]: here

File: index.d.ts [hunk 2]
```diff
@@ -683,56 +718,78 @@ declare module 'plaid' {
...
+      products: Array<string>,
+      userAuth: Map<string,string>,
+      cb: Callback<TokenResponse>,
+    ): void;
+    importItem(
+      products: Array<string>,
+      userAuth: Map<string, string>,
+      options: WebhookOptions,
+      cb: Callback<TokenResponse>,
+    ): void;
```
Comment (jeffzwang) [inline index.d.ts]: here

File: index.d.ts [hunk 3]
```diff
@@ -623,6 +629,28 @@ declare module 'plaid' {
...
+
+  interface DepositSwitchCreateResponse extends BaseResponse {
+    deposit_switch_id: string;
+  }
+
+  interface DepositSwitchTokenCreateResponse extends BaseResponse {
+    deposit_switch_token: string;
+    deposit_switch_token_expiration_time: string;
+  }
+
```
Comment (jeffzwang) [inline index.d.ts]: here

File: index.d.ts [hunk 4]
```diff
@@ -52,6 +52,10 @@ declare module 'plaid' {
 
   type AssetReportRefreshOptions = AssetReportCreateOptions;
 
+  interface WebhookOptions {
+    webhook?: string;
+  }
+
```
Comment (jeffzwang) [inline index.d.ts]: here

File: index.d.ts [hunk 5]
```diff
@@ -52,6 +52,10 @@ declare module 'plaid' {
 
   type AssetReportRefreshOptions = AssetReportCreateOptions;
 
+  interface WebhookOptions {
```
Comment (jeffzwang) [inline index.d.ts]: Please ignore all lines except for the ones that I have marked with comments saying "here". I just prettified things

File: test/PlaidClientTest.js [hunk 6]
```diff
@@ -937,6 +973,84 @@ describe('plaid.Client', () => {
...
+      };
+
+      const getAccountId = (access_token, cb) => {
+        pCl.getAccounts(access_token, (err, response) => {
+          expect(err).to.be(null);
+          expect(response).to.be.ok();
+          expect(response.accounts).to.be.ok();
+          cb(null,
+            {
+               account_id: response.accounts[0].account_id,
```
Comment (rayraycano) [inline test/PlaidClientTest.js]: could you perhaps filter for account type/subtype instead of using the 0th index?

Comment (jeffzwang) [reply test/PlaidClientTest.js]: ⚡️

File: index.d.ts [hunk 7]
```diff
@@ -683,56 +718,78 @@ declare module 'plaid' {
 
     getItem: AccessTokenFn<ItemResponse>;
 
-    updateItemWebhook(accessToken: string,
-                      webhook: string,
+    importItem(
+      products: Array<string>,
+      userAuth: Map<string,string>,
+      options?: WebhookOptions,
+    ): void;
```
Comment (skansal-plaid) [inline index.d.ts]: `s/void/Promise<ImportItemResponse>`

File: index.d.ts [hunk 8]
```diff
@@ -683,56 +718,78 @@ declare module 'plaid' {
...
+    importItem(
+      products: Array<string>,
+      userAuth: Map<string,string>,
+      cb: Callback<TokenResponse>,
```
Comment (skansal-plaid) [inline index.d.ts]: `s/Callback<TokenResponse>/Callback<ImportItemResponse>` unless you meant to re-use `TokenResponse` here

File: index.d.ts [hunk 9]
```diff
@@ -683,56 +718,78 @@ declare module 'plaid' {
...
+    ): void;
+    importItem(
+      products: Array<string>,
+      userAuth: Map<string, string>,
+      options: WebhookOptions,
+      cb: Callback<TokenResponse>,
```
Comment (skansal-plaid) [inline index.d.ts]: https://github.com/plaid/plaid-node/pull/237/files#r372061532

File: index.d.ts [hunk 10]
```diff
@@ -744,179 +801,290 @@ declare module 'plaid' {
...
+      endDate: Iso8601DateString,
+      options: GetAllTransactionsRequestOptions,
+      cb: Callback<TransactionsAllResponse>,
+    ): void;
+
+    // getDepositSwitch(String, Object?, Function)
+    getDepositSwitch(
+      depositSwitchId: string,
+      options?: Object,
+    ): void;
```
Comment (skansal-plaid) [inline index.d.ts]: return promise here?

File: index.d.ts [hunk 11]
```diff
@@ -744,179 +801,290 @@ declare module 'plaid' {
...
+      options: Object,
+      cb: Callback<DepositSwitchGetResponse>,
+    ): void;
+
+    // createDepositSwitch(String, String, Object?, Function)
+    createDepositSwitch(
+      targetAccountId: string,
+      targetAccessToken: string,
+      options?: Object,
+    ): void;
```
Comment (skansal-plaid) [inline index.d.ts]: return promise here?

File: index.d.ts [hunk 12]
```diff
@@ -744,179 +801,290 @@ declare module 'plaid' {
...
+      targetAccessToken: string,
+      options: Object,
+      cb: Callback<DepositSwitchCreateResponse>,
+    ): void;
+
+    // createDepositSwitchToken(String, Object?, Function)
+    createDepositSwitchToken(
+      depositSwitchId: string,
+      options?: Object,
+    ): void;
```
Comment (skansal-plaid) [inline index.d.ts]: return promise here?

File: index.d.ts (hunk 9 above)
Comment (jeffzwang) [reply index.d.ts]: ⚡️

File: index.d.ts (hunk 10 above)
Comment (jeffzwang) [reply index.d.ts]: ⚡️

File: index.d.ts (hunk 11 above)
Comment (jeffzwang) [reply index.d.ts]: ⚡️

File: index.d.ts (hunk 12 above)
Comment (jeffzwang) [reply index.d.ts]: ⚡️

**Pull request merged**